common/lockmanager_test.go

```diff
@@ -0,0 +1,114 @@
+package common
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/edulinq/autograder/config"
+)
+
+var (
+    key = "testkey1"
+    key2 = "testkey2"
+    doesNotExistKey = "dne"
+    staleDuration = time.Duration(config.STALELOCK_DURATION.Get()) * time.Second;
+    wg sync.WaitGroup;
+)
+
+func TestLockManager(test *testing.T) {
+    testLockingKey()
+    
+    testUnlockingKey(test)
+
+    testConcurrentLockingUnlocking()
+
+    testUnlockingAnUnlockedLock(test)
+    
+    testUnlockingMissingKey(test)
+
+    testLockConcurrencyWithStaleCheck(test)
+}
+
+func testLockingKey() {
+    Lock(key)
+}
+
+func testUnlockingKey(test *testing.T) {
+    err := Unlock(key);
+    if (err != nil) {
+        test.Errorf("Failed to unlock a key");
+    }
+}
+
+func testConcurrentLockingUnlocking() {
```

Comment:
For difficult tests like this, put in some comments to let people know what you are doing.